File: src/main/java/com/timepath/util/concurrent/DaemonThreadFactory.java
package com.timepath.util.concurrent;

import org.jetbrains.annotations.NotNull;

import java.util.concurrent.Executors;
import java.util.concurrent.ThreadFactory;

/**
 * @author TimePath
 */
public class DaemonThreadFactory implements ThreadFactory {

    private final ThreadFactory threadFactory = Executors.defaultThreadFactory();

    @Override
    public Thread newThread(@NotNull Runnable r) {
        Thread t = threadFactory.newThread(r);
        t.setDaemon(true);
        return t;
    }
}

File: src/main/kotlin/com/timepath/util/concurrent/DaemonThreadFactory.kt
package com.timepath.util.concurrent


import java.util.concurrent.Executors
import java.util.concurrent.ThreadFactory

public class DaemonThreadFactory : ThreadFactory {

    private val threadFactory = Executors.defaultThreadFactory()

    override fun newThread(r: Runnable) = threadFactory.newThread(r).apply { isDaemon = true }
}
